src/org/traccar/protocol/Jt600ProtocolDecoder.java

```diff
@@ -256,11 +259,29 @@ private Position decodeU01(String sentence, Channel channel, SocketAddress remot
         position.set(Position.KEY_CID, parser.nextInt());
         position.set(Position.KEY_LAC, parser.nextInt());
         position.set(Position.KEY_GSM, parser.nextInt());
-        position.set(Position.KEY_ODOMETER, parser.nextLong() * 1000);
+        position.set(Position.KEY_ODOMETER, parser.nextLong() * 100);
         position.set(Position.KEY_INDEX, parser.nextInt());
 
-        return position;
+        switch (messageType) {
+            case "U01":
+            case "U02":
+            case "U03":
+                String checkSum = parser.next();
+                int calculatedCheckSum = checkSum(sentence.substring(1, sentence.length() - checkSum.length() - 1));
```

Comment:
Do we really need to check a checksum? I don't do it for other protocols because it seems redundant as TCP and UDP protocols already have 16-bit checksum which is more reliable than 1-byte xor checksum here.